### PR TITLE
Fix tests parallelism

### PR DESCRIPTION
**Pull request by toninorair**
The real fix for tests parallelism 
instead of changing validator env vars just to get signature in a convenient way, this approach gets signature directly.

### CODE REVIEW

File: pallets/cash/src/tests/mod.rs [hunk 1]
```diff
@@ -159,11 +160,12 @@ pub fn initialize_storage() {
...
-        "ETH_KEY",
-        "6bc5ea78f041146e38233f5bc29c703c1cec8eaaa2214353ee8adf7fc598f23d",
-    );
-    validator_sign::<Test>(data)
+    let private_key =
+        hex::decode("6bc5ea78f041146e38233f5bc29c703c1cec8eaaa2214353ee8adf7fc598f23d").unwrap();
+    let secret_key = SecretKey::parse_slice(&private_key).unwrap();
+    let signature_raw = gateway_crypto::eth_sign(data, &secret_key, false);
+    let signature = ChainSignature::Eth(signature_raw);
+    Ok(signature)
```
Comment (jflatow) [inline pallets/cash/src/tests/mod.rs:177]: Hm this doesn't exercise `validator_sign` anymore though, but maybe that's ok if it's tested elsewhere?

Comment (toninorair) [reply pallets/cash/src/tests/mod.rs:177]: It returns signature, I think that's actually the right way to do it. To get a validator signature, there is no need to change env var for the convenience of calling `validator_sign`.  That was casing tests failures unless they all run serially, now I see all tests passing

Comment (toninorair) [reply pallets/cash/src/tests/mod.rs:177]: @jflatow do you think there is a need to change env var?

Comment (jflatow) [reply pallets/cash/src/tests/mod.rs:177]: I think that makes sense, its just like how do we test `validator_sign`?

Comment (toninorair) [reply pallets/cash/src/tests/mod.rs:177]: should it be tested at unit level? it realistically just the wrapper around sign_message --> eth_sign. It's like if we want to run tests in parallel then env vars should not be changed. 
Alternatively tests that change vars can be marked with some special attribute and run separately. But the whole test suite, there is only one such test now.

**Comment by toninorair** [comment]
@jflatow let's merge it in? It is an improvement anyway and allows to run tests in parallel

**Comment by jflatow** [comment]
I don't think this fixes the issue, closing

**Comment by toninorair** [comment]
> I don't think this fixes the issue, closing

but it did, it was actually fixing the issue with running tests in parallel by getting signature without changing the env var

**Comment by toninorair** [comment]
It's out of sync now, I will clean it up.

**Comment by jflatow** [comment]
OK I rebased and can confirm this does seem to work now, but doesn't exercise the `validator_sign` path, so let's fix again as we try to get rid of the env var completely next

**Comment by jflatow** [comment]
Sorry 🎠, this got closed when I deleted the base moon branch and now I can't reopen, opening another PR

**Pull request closed without merging**